### PR TITLE
Remove Doctrine Events

### PR DESCRIPTION
**Pull request by HedicGuibert**
This completely removes Doctrine Events.

I also prevented notifications from being sent if a submit status is changed to done, because it wasn't handled well and it's probably not really interresting.

### CODE REVIEW

File: src/EventListener/SlackNotifierEventListener.php [hunk 1]
```diff
@@ -117,4 +125,19 @@ public function onCfpEndingSoon(CfpEndingSoonEvent $event): void
         $payload['attachments'][] = $event->buildAttachment();
         $this->slackNotifier->notify($payload);
     }
+
+    /**
+     * @param GenericEvent<EasyAdminEvents> $event
```
Comment (lyrixx) [inline src/EventListener/SlackNotifierEventListener.php:130]: Do this works? What does it mean?

File: src/EventListener/SlackNotifierEventListener.php [hunk 2]
```diff
@@ -117,4 +125,19 @@ public function onCfpEndingSoon(CfpEndingSoonEvent $event): void
...
+    {
+        $entity = $event->getSubject();
+        if ($entity instanceof Submit) {
+            $statusChanged = $entity->getStatusChanged();
+            if (null !== $statusChanged) {
+                $this->eventDispatcher->dispatch(new SubmitStatusChangedEvent($entity));
+                $entity->resetStatusChanged();
+            }
+        }
+    }
```
Comment (lyrixx) [inline src/EventListener/SlackNotifierEventListener.php:142]: This code does not belong to this class. It has nothing related to slack :)

File: src/EventListener/DoctrineEventSubscriber.php [hunk 3]
```diff
@@ -1,88 +0,0 @@
...
-    }
-
-    public function postFlush(): void
-    {
-        if (1 === \count($this->conferencesAdded)) {
-            $this->eventDispatcher->dispatch(new NewConferenceEvent($this->conferencesAdded[0], $this->router));
-        } elseif (\count($this->conferencesAdded) > 1) {
-            $this->eventDispatcher->dispatch(new NewConferencesEvent($this->conferencesAdded, $this->router));
-        }
-    }
```
Comment (lyrixx) [inline src/EventListener/DoctrineEventSubscriber.php:87]: I failed to see where this code is now

File: src/EventListener/SlackNotifierEventListener.php (hunk 1 above)
Comment (HedicGuibert) [reply src/EventListener/SlackNotifierEventListener.php]: It works, it is just for phpstan. Since GenericEvent is an iterable, phpstan wants us to tell the iterable value type.
We could replace it with a phpstan-ignore-next-line but I think it is worse.

File: src/EventListener/DoctrineEventSubscriber.php (hunk 3 above)
Comment (HedicGuibert) [reply src/EventListener/DoctrineEventSubscriber.php:87]: It is in the ConferenceHarvester, lines 98-102

File: src/EventListener/SlackNotifierEventListener.php (hunk 1 above)
Comment (lyrixx) [reply src/EventListener/SlackNotifierEventListener.php]: Ok ok 👍🏼 (you don't iterate on it, but not pb :) )

File: src/Entity/Submit.php [hunk 4]
```diff
@@ -171,4 +174,14 @@ public function reduceSpeakersNames(): string
             return '' === $r ? $u->getName() : $r.', '.$u->getName();
         }, '');
     }
+
+    public function getStatusChanged(): bool
```
Comment (lyrixx) [inline src/Entity/Submit.php:178]:
```suggestion
    public function isStatusChanged(): bool
```
no?

File: src/Event/NewConferenceEvent.php [hunk 5]
```diff
@@ -87,7 +85,7 @@ public function buildAttachment(): array
         if ($this->conference->getParticipations()->count() > 0) {
             $starfleetLinkField = SlackNotifier::SHORT_FIELD;
             $starfleetLinkField['title'] = 'Starfleet link  🚀';
-            $starfleetLinkField['value'] = $this->router->generate('conferences_show', [
+            $starfleetLinkField['value'] = $router->generate('conferences_show', [
```
Comment (lyrixx) [inline src/Event/NewConferenceEvent.php:88]: Ohhhh, I see why the router was in this class before.
Don't modify this code for now, but It is absolutely not the place for this method :) 
It's not the responsibility to the event to know how to transform it to a slack attachment. It's the responsibility of the slack notifier.

TL;DR: don't do it now, but theses method should be move to the slack notifier (you can do it later, or not, up to you :) )

**Review by lyrixx — APPROVED** [review]
Very good job. Thanks

File: src/Entity/Submit.php (hunk 4 above)
Comment (lyrixx) [reply src/Entity/Submit.php:178]: Could be done later, or not :)
